src/context/authContext/index.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "../../firebase/firebase";
+import { useContext, useEffect, useState } from "react";
+
+const AuthContext = React.createContext();
+
+export function useAuth() {
+  return useContext(AuthContext);
+}
+
+export function AuthProvider({ children }) {
+  const [currentUser, setCurrentUser] = useState(null);
```

Comment:
What info does this state contain? Please update the variable names accordingly. If it's a token, it can be named userToken for example.